File: src/main/java/org/jcodec/containers/mp4/boxes/SoundMediaHeaderBox.java
package org.jcodec.containers.mp4.boxes;

import js.nio.ByteBuffer;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * Sound media header
 * 
 * @author The JCodec project
 * 
 */
public class SoundMediaHeaderBox extends FullBox {
    private short balance;

    public static String fourcc() {
        return "smhd";
    }

    public static SoundMediaHeaderBox createSoundMediaHeaderBox() {
        return new SoundMediaHeaderBox(new Header(fourcc()));
    }

    public SoundMediaHeaderBox(Header atom) {
        super(atom);
    }

    public void parse(ByteBuffer input) {
        super.parse(input);
        balance = input.getShort();
        input.getShort();
    }

    protected void doWrite(ByteBuffer out) {
        super.doWrite(out);
        out.putShort(balance);
        out.putShort((short) 0);
    }

    public short getBalance() {
        return balance;
    }
}

File: src/main/java/org/jcodec/containers/mp4/boxes/SoundMediaHeaderBox.kt
package org.jcodec.containers.mp4.boxes

import java.nio.ByteBuffer

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * Sound media header
 *
 * @author The JCodec project
 */
class SoundMediaHeaderBox(atom: Header) : FullBox(atom) {
    var balance: Short = 0
        private set

    override fun parse(input: ByteBuffer) {
        super.parse(input)
        balance = input.short
        input.short
    }

    override fun doWrite(out: ByteBuffer) {
        super.doWrite(out)
        out.putShort(balance)
        out.putShort(0.toShort())
    }

    override fun estimateSize(): Int {
        return 16
    }

    companion object {
        @JvmStatic
        fun fourcc(): String {
            return "smhd"
        }

        @JvmStatic
        fun createSoundMediaHeaderBox(): SoundMediaHeaderBox {
            return SoundMediaHeaderBox(Header(fourcc()))
        }
    }
}